viewing_party/party.py

```diff
@@ -1,23 +1,176 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie
+    else: 
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    wacthlist = user_data["watchlist"]
+
+    for movie in wacthlist:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    """
+    Calculates the average rating of watched movies.
+
+    Returns:
+    float: The average rating of watched movies, or 0.0 if none are watched.
+    """
+
+    avg_rating = 0.0
+    count = 0
+    sum = 0
+
+    if not user_data["watched"]:
+        return avg_rating
+    
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+        count += 1
 
+    avg_rating = sum / count

```

Comment:
Do we need to create a variable to track `count` and then manually increment it ourselves? 

If there is a built-in function that can be used then I'd prefer that so that we can reduce the places a bug could get introduced.
```suggestion
    for movie in user_data["watched"]:
        sum += movie["rating"]

    avg_rating = sum / len(user_data["watched"])
```